datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -53,6 +53,13 @@ abstract class DefaultMatchPattern extends MatchPattern[ModularPlan] {
             (a.child.asInstanceOf[Attribute], a.toAttribute)
           })
 
+    // Create aliasMap with Cast Expression to alias reference attribute
+    val aliasMapCast =
+      subsumer.outputList.collect {
+        case a: Alias if a.child.isInstanceOf[Cast] =>
```

Comment:
i think, you cannot just restrict it to cast, because cast is in this case, for example, if i just have query as 
`select floor(m_month) as a, c_code as abc from maintable` (query derived from your test case) , then this also fails. Because here you will have project as `[FLOOR(cast(age#7 as double)) AS a#44L]`. So again this scenario needs to be handled in generalized way i  think

Reply:
okay